pkgs/test_core/lib/src/runner/reporter/json.dart

```diff
@@ -178,7 +178,8 @@ class JsonReporter implements Reporter {
     // different metadata.
     if (suite is LoadSuite) {
       suite.suite.then((runnerSuite) {
-        _suiteIDs[runnerSuite!] = id;
+        if (runnerSuite == null) return;
```

Comment:
oops, looks like I never merged this back to the `null_safety` branch.

I did not think this was a situation we wanted to support.

https://github.com/dart-lang/test/pull/1245

Do you have an understanding of how this shows up as null here?

Reply:
I do not, no

Reply:
From the docs:

```
/// A future that completes to the loaded suite once the suite's test has been
  /// run and completed successfully.
  ///
  /// This will return `null` if the suite is unavailable for some reason (for
  /// example if an error occurred while loading it).
```